Source/Scene/PostProcessAmbientOcclusionStage.js

```diff
@@ -0,0 +1,199 @@
+define([
+        '../Core/Check',
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/destroyObject',
+        '../Core/PixelFormat',
+        '../Renderer/PixelDatatype',
+        '../Renderer/Sampler',
+        '../Renderer/Texture',
+        '../Renderer/TextureMagnificationFilter',
+        '../Renderer/TextureMinificationFilter',
+        '../Renderer/TextureWrap',
+        '../Shaders/PostProcessFilters/AmbientOcclusion',
+        '../Shaders/PostProcessFilters/AmbientOcclusionGenerate',
+        './PostProcess',
+        './PostProcessBlurStage',
+        './PostProcessComposite'
+    ], function(
+        Check,
+        defined,
+        defineProperties,
+        destroyObject,
+        PixelFormat,
+        PixelDatatype,
+        Sampler,
+        Texture,
+        TextureMagnificationFilter,
+        TextureMinificationFilter,
+        TextureWrap,
+        AmbientOcclusion,
+        AmbientOcclusionGenerate,
+        PostProcess,
+        PostProcessBlurStage,
+        PostProcessComposite) {
+    'use strict';
+
+    /**
+     * Post process stage for ambient occlusion. Implements {@link PostProcess}.
```

Comment:
It may be easier to discuss offline, but should this implement `PostProcess`?  Perhaps a better architecture is a factory that returns a new composite post process?